lib/internal/url.js

```diff
@@ -1265,6 +1265,10 @@ function domainToUnicode(domain) {
   return _domainToUnicode(`${domain}`);
 }
 
+function decodeAuth(str) {
+  return decodeURIComponent(str).replace(':', '%3A').replace('/', '%2F');
```

Comment:
https://github.com/addaleax/node/blob/5ebbb79a3bd56b99821081c23e1f5db9c879fce8/lib/internal/url.js#L1296

```diff
const forwardSlashRegEx = /\//g;
+const colonRegEx = /:/g;
```

```suggestion
  return decodeURIComponent(str).replace(colonRegEx, '%3A').replace(forwardSlashRegEx, '%2F');
```

Reply:
This is the only thing `todo` here I think...